pkg/buildpacks/telemetry.go

```diff
@@ -0,0 +1,155 @@
+package buildpacks
+
+import (
+	"encoding/json"
+
+	"github.com/SAP/jenkins-library/pkg/cnbutils/privacy"
+	"github.com/SAP/jenkins-library/pkg/cnbutils/project"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+const version = 3
+
+type Telemetry struct {
+	customData *telemetry.CustomData
+	data       *BuildpacksTelemetry
+}
+
+func NewTelemetry(customData *telemetry.CustomData) Telemetry {
+	return Telemetry{
+		customData: customData,
+		data: &BuildpacksTelemetry{
+			Version: version,
+		},
+	}
+}
+
+func (d *Telemetry) Export() error {
+	d.customData.Custom1Label = "cnbBuildStepData"
+	customData, err := json.Marshal(d.data)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal custom telemetry data")
+	}
+	d.customData.Custom1 = string(customData)
+	return nil
+}
+
+func (d *Telemetry) WithImage(image string) {
+	d.data.builder = image
+}
+
+func (d *Telemetry) AddSegment(segment *Segment) {
+	segment.data.Builder = d.data.builder
+	d.data.Data = append(d.data.Data, segment.data)
+}
+
+type BuildpacksTelemetry struct {
+	builder string
+	Version int                      `json:"version"`
+	Data    []*cnbBuildTelemetryData `json:"data"`
+}
+
+type cnbBuildTelemetryData struct {
+	ImageTag          string                                 `json:"imageTag"`
+	AdditionalTags    []string                               `json:"additionalTags"`
+	BindingKeys       []string                               `json:"bindingKeys"`
+	Path              PathEnum                               `json:"path"`
+	BuildEnv          cnbBuildTelemetryDataBuildEnv          `json:"buildEnv"`
+	Buildpacks        cnbBuildTelemetryDataBuildpacks        `json:"buildpacks"`
+	ProjectDescriptor cnbBuildTelemetryDataProjectDescriptor `json:"projectDescriptor"`
+	BuildTool         string                                 `json:"buildTool"`
+	Builder           string                                 `json:"builder"`
+}
+
+type cnbBuildTelemetryDataBuildEnv struct {
+	KeysFromConfig            []string               `json:"keysFromConfig"`
+	KeysFromProjectDescriptor []string               `json:"keysFromProjectDescriptor"`
+	KeysOverall               []string               `json:"keysOverall"`
+	JVMVersion                string                 `json:"jvmVersion"`
+	KeyValues                 map[string]interface{} `json:"keyValues"`
+}
+
+type cnbBuildTelemetryDataBuildpacks struct {
+	FromConfig            []string `json:"FromConfig"`
+	FromProjectDescriptor []string `json:"FromProjectDescriptor"`
+	Overall               []string `json:"overall"`
+}
+
+type cnbBuildTelemetryDataProjectDescriptor struct {
+	Used        bool `json:"used"`
+	IncludeUsed bool `json:"includeUsed"`
+	ExcludeUsed bool `json:"excludeUsed"`
+}
+
+type Segment struct {
+	data *cnbBuildTelemetryData
+}
+
+func NewSegment() *Segment {
+	return &Segment{
+		data: &cnbBuildTelemetryData{},
+	}
+}
+
+func (s *Segment) WithBindings(bindings map[string]interface{}) {
```

Comment:
Same as above, but for the `TelemetrySegment`, we could actually use the fluent API even.